src/repr/src/scalar.rs

```diff
@@ -2757,6 +2831,119 @@ fn arb_list(element_strategy: BoxedStrategy<PropDatum>) -> BoxedStrategy<PropLis
         .boxed()
 }
 
+#[derive(Debug, PartialEq, Clone)]
+pub struct PropRange(
+    Row,
+    Option<(
+        (Option<Box<PropDatum>>, bool),
+        (Option<Box<PropDatum>>, bool),
+    )>,
+);
+
+fn arb_range_data() -> BoxedStrategy<(PropDatum, PropDatum)> {
```

Comment:
This does not make a lot of sense. You can inline 

```rust
any::<i32>().prop_map(PropDatum::Int32),
any::<i32>().prop_map(PropDatum::Int32),
```

in the `arb_range` strategy list instead of `arb_range_data(),` and pattern match `upper_inc, a, b` in the `prop_map` call.

Reply:
I am going to be adding additional data types in a subsequent commit (have it locally just not in this PR), and I need something that constrains that they're the same type and one of the subtypes that ranges support. Wasn't sure how else to enforce those constraints.